__tests__/test.js

```diff
@@ -51,6 +56,10 @@ it('basic query', () => {
     return testGetFields('{ someType { a b } }', ['a', 'b']);
 });
 
+it('get fields on scalar field', () => {
+    return testGetFields('{ someType { a } }', [], {}, true);
```

Comment:
Why is this expected to return `[]`? The `a` field is being queried on `someType`, why would it not be returned?

Reply:
Ohh, you're trying to do this on the 'a' field itself.

Reply:
I added the extra argument `getFieldsForSubfield` to specify that we want this run of `testGetFields()` to call `getFieldList()` on the subfield of the given query.

Reply:
Can you instead add the scalar to the main QueryType as `s` or something? Then you can change the test call to ` testGetFields('{ s }', [])` and not need to add the getFieldsForSubfield arg.

Reply:
Oh good point, that's much better. I'll do that now 👍